dali/test/python/test_operator_random_object_bbox.py

```diff
@@ -0,0 +1,364 @@
+            assert contains_box(cls_boxes, boxes[i])
+
+
+def test_random_object_bbox_with_class():
+    np.random.seed(12345)
+    types = [np.int8, np.uint8, np.int16, np.uint16, np.int32, np.uint32]
+
+    formats = [None, "anchor_shape", "start_end", "box"]
+    fmt = 0
+    for bg in [None, 0, -1, 5, fn.random.uniform(range=(-5, 10), dtype=dali.types.INT32, seed=12321)]:
+        if bg is None or isinstance(bg, int):
+            class_opt = [None, [0], [1], [2,4,5,7]]
+            for x in class_opt:
+                if isinstance(x, list) and bg in x:
+                    x.remove(bg)
+            if [] in class_opt: class_opt.remove([])
+            # putting this in the list interefered with remove
+            class_opt.append(random_classes(0 if bg is None else bg))
+        else:
+            class_opt = [None]
+        for classes in class_opt:
+            if classes is None:
+                weights_opt = [None, [1], [0.5, 1, 0.1, 0.2], random_weights()]
+            elif isinstance(classes, list):
+                weights_opt = [None, list(range(1, 1+len(classes)))]
+            else:
+                weights_opt = [None]
+
+            for weights in weights_opt:
+                ndim = np.random.randint(1, 5)
+
+                threshold_opt = [None, 3, list(range(1, 1+ndim)), fn.random.uniform(range=(1,5), shape=[ndim], dtype=dali.types.INT32, seed=13231)]
+                threshold = threshold_opt[np.random.randint(len(threshold_opt))]
+                k_largest_opt = [None, 1, 2, 5]
+                k_largest = k_largest_opt[np.random.randint(len(k_largest_opt))]
+
+                fg_prob_opt = [None, 0.1, 0.7, fn.random.uniform(range=(0,1), seed=1515)]
+                fg_prob = fg_prob_opt[np.random.randint(len(fg_prob_opt))]
+
+                format = formats[fmt]
+                fmt = (fmt + 1) % len(formats)
+                dtype = types[np.random.randint(0, len(types))]
+                yield _test_random_object_bbox_with_class, 5, ndim, dtype, format, fg_prob, classes, weights, bg, threshold, k_largest
+
+@nottest
+def _test_random_object_bbox_ignore_class(max_batch_size, ndim, dtype, format=None, background=None, threshold=None, k_largest=None):
+    pipe = dali.pipeline.Pipeline(max_batch_size, 4, device_id = None, seed=4321)
+    background_out = 0 if background is None else background
+    threshold_out = np.int32([]) if threshold is None else threshold
+
+    with pipe:
+        inp = fn.external_source(batch_generator(max_batch_size, ndim, dtype))
+        outs = fn.segmentation.random_object_bbox(inp, format=format, ignore_class=True, background=background, seed=1234,
+                                                  threshold=threshold, k_largest=k_largest)
+        if not isinstance(outs, list):
+            outs = [outs]
+        pipe.set_outputs(inp, background_out, threshold_out, *outs)
+    pipe.build()
+
+    format = format or "anchor_shape"
+
+    for _ in range(50):
```

Comment:
can you make the number of iterations a parameter of the function?

Reply:
I could, but what for?

Reply:
It's easy to change later if we need to